### ISSUE TITLE
Global map of trends (in each grid cell)

### ISSUE THREAD

**Issue by mara-freilich** [mfreilich1/cmip6hack-ocean-bgc#4]

**Comment by bradyrx** [comment]
@mfreilich1 , I added this to the slack, but I wrote up a gist to demonstrate how to do this with dask and xarray to do in parallel at every grid cell for a full ensemble (https://nbviewer.jupyter.org/gist/bradyrx/41e4fa86a92908deecd422503b62a29b). It can handle 11 members of CESM2 in 15 seconds. 

I've been working on this for awhile and think `apply_ufunc` works best. But it's a common problem in climate science to need to do these vectorized options over a grid and it's not always an easy solution. I also have code written like this to remove polynomial fits, etc. if that's needed as well.

**Comment by mara-freilich** [comment]
Cool! Thanks! This is really helpful. I'll use this is a model for plotting some of the other variables we're interested in!

**Comment by bradyrx** [comment]
No problem. Let me know if you have any trouble with adapting it to other common functions too like removing a trend or polynomial at each grid cell.

**Comment by jbusecke** [comment]
Great work @bradyrx . I adopted a similar approach (apply_ufunc FTW!) but with scipy‘s linregress function, which gives some more parameters ( p value etc). [Here](https://github.com/jbusecke/xarrayutils/blob/master/xarrayutils/utils.py ) is an example how to wrap that neatly into an xarray (`xr_linregress` is the function to check out)

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by jbusecke** [comment]
The package is mostly not documented...I should do that but this function is tested well!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by bradyrx** [comment]
Thanks @jbusecke! To be clear, this is usually the approach I take too. I find `linregress` to be a bit slower with `apply_ufunc` since it has all the extra bells and whistles. Raw numpy always seems to be a little faster, so I use that when I can. I've got a version where I took the raw numpy from `scipy.linregress` and stripped that down just to get slope + p-value as well. Anyways, agreed `apply_ufunc` FTW once you figure out all the kwargs!
